### ISSUE TITLE
Waves 1.1.5 stuck on fork

### ISSUE THREAD

**Issue by ksemaev** [wavesplatform/Waves#2709]
Got my waves-node stuck on block `1783307`

The error is:

```
2019-11-06 12:47:02,739 INFO  [nio-worker-group-3-6] c.w.n.HandshakeHandler$Client - [a597488e 52.52.46.76:6868] Accepted handshake Handshake(wavesW,(1,1,5),waves-official-eu-node-4,1210162,Some(/52.52.46.76:6868))
2019-11-06 12:47:08,892 WARN  [appender-56] c.w.s.a.ExtensionAppender$ - Can't process fork starting with 4Y4yroLLjYPkXML65cGG59eCMAidNj6LoVqtMSdAw6ABfpzwGXRFLnsvr1qTzPiKKXH3u8qUfotkmnt7ZRhEFPQv, error appending block Block(3xGGgXJzGYAQxmL3s8XNsxp68m4YsCqPSNYafrJaAUPs5yox4GNH2AJjL4TLAEciM4L5GUvJBwzcNHJeD8CtjNdF -> 4Y4yroL..., txs=116, features=Set()): TransactionValidationError(cause = AccountBalanceError(Map(3PQxNpso2uNbiPM7PQWJMNeYkVsUv4P5mLm -> negative asset balance: 3PQxNpso2uNbiPM7PQWJMNeYkVsUv4P5mLm, new portfolio: Map(4LHHvYGNKJUg5hj65aGD5vgScvCBmLpdRFtjokvCjSL8 -> -113898165))),
tx = {
  "senderPublicKey" : "E3UwaHCQCySghK3zwNB8EDHoc3b8uhzGPFz3gHmWon4W",
  "amount" : 113898165,
  "fee" : 1400000,
  "type" : 7,
  "version" : 2,
  "sellMatcherFee" : 699999,
  "sender" : "3PJjwFREg8F9V6Cp9fnUuEwRts6HQQa5nfP",
  "feeAssetId" : null,
  "proofs" : [ "d9FPjDveSHU5cJTgKDxfT3v6t29PcTg4r3TkAYUHuhxFur1NxCqENkYUe6gKgowFhvjT2MKBQV17bc32tcdEN8Q" ],
  "price" : 11636184,
  "id" : "9tLCF4omb9hWmVXHwbj4CkiTkNDyTbC9xCmw65D1MKS3",
  "order2" : {
    "version" : 2,
    "id" : "DB2eX7VHJSkBvC95FYHWgEvHj47EsReHN4LQnD1EMLnb",
    "sender" : "3PQxNpso2uNbiPM7PQWJMNeYkVsUv4P5mLm",
    "senderPublicKey" : "Cb92PCaMeimQmmK1dSaoUro3j8YhskKhH89JWdmsCkQX",
    "matcherPublicKey" : "E3UwaHCQCySghK3zwNB8EDHoc3b8uhzGPFz3gHmWon4W",
    "assetPair" : {
      "amountAsset" : "4LHHvYGNKJUg5hj65aGD5vgScvCBmLpdRFtjokvCjSL8",
      "priceAsset" : null
    },
    "orderType" : "sell",
    "amount" : 113898170,
    "price" : 11636184,
    "timestamp" : 1572775953909,
    "expiration" : 1575281553909,
    "matcherFee" : 700000,
    "signature" : "2SY4WTcKVqLX97jnLC2iGghv2Q759S1HYWkHqueNUcVnSAH2YHt7F4Lr5onAHRA5LkJMAHYjw8byGiUmNG14ixXc",
    "proofs" : [ "2SY4WTcKVqLX97jnLC2iGghv2Q759S1HYWkHqueNUcVnSAH2YHt7F4Lr5onAHRA5LkJMAHYjw8byGiUmNG14ixXc" ]
  },
  "order1" : {
    "version" : 2,
    "id" : "HfK4b193jVSLuSZNFbgz1cSdN2qa1HkMJvEB86dFQCDY",
    "sender" : "3PAn2voDZPPiqgXPXNYENYit8PeEgGqey2A",
    "senderPublicKey" : "4EdEM9rL9mjZn3drNnkceH6DFVdYUgPx9jEmHN9aqEiw",
    "matcherPublicKey" : "E3UwaHCQCySghK3zwNB8EDHoc3b8uhzGPFz3gHmWon4W",
    "assetPair" : {
      "amountAsset" : "4LHHvYGNKJUg5hj65aGD5vgScvCBmLpdRFtjokvCjSL8",
      "priceAsset" : null
    },
    "orderType" : "buy",
    "amount" : 8544502410,
    "price" : 11682774,
    "timestamp" : 1573040813651,
    "expiration" : 1575546413651,
    "matcherFee" : 300000,
    "signature" : "3s2NmG4DzvwtXB3ZFm9JsYUUzH5ZV3qDhHeCdCrXCcZxf4z74PADf8AMogWdkU2GKda7WKJrNZ68jKiis2gmXzJR",
    "proofs" : [ "3s2NmG4DzvwtXB3ZFm9JsYUUzH5ZV3qDhHeCdCrXCcZxf4z74PADf8AMogWdkU2GKda7WKJrNZ68jKiis2gmXzJR" ]
  },
  "buyMatcherFee" : 3998,
  "timestamp" : 1573040813705
})
2019-11-06 12:47:08,893 INFO  [appender-56] c.w.s.BlockchainUpdaterImpl - Removing blocks after 4Y4yroL... from blockchain
2019-11-06 12:47:08,909 WARN  [rx-extension-loader-40] c.w.s.appender.package$ - [a597488e 52.52.46.76:6868] Error appending extension (total=56) [3xGGgXJ... -- 6318Xg3...]: TransactionValidationError(cause = AccountBalanceError(Map(3PQxNpso2uNbiPM7PQWJMNeYkVsUv4P5mLm -> negative asset balance: 3PQxNpso2uNbiPM7PQWJMNeYkVsUv4P5mLm, new portfolio: Map(4LHHvYGNKJUg5hj65aGD5vgScvCBmLpdRFtjokvCjSL8 -> -113898165))),
tx = {
  "senderPublicKey" : "E3UwaHCQCySghK3zwNB8EDHoc3b8uhzGPFz3gHmWon4W",
  "amount" : 113898165,
  "fee" : 1400000,
  "type" : 7,
  "version" : 2,
  "sellMatcherFee" : 699999,
  "sender" : "3PJjwFREg8F9V6Cp9fnUuEwRts6HQQa5nfP",
  "feeAssetId" : null,
  "proofs" : [ "d9FPjDveSHU5cJTgKDxfT3v6t29PcTg4r3TkAYUHuhxFur1NxCqENkYUe6gKgowFhvjT2MKBQV17bc32tcdEN8Q" ],
  "price" : 11636184,
  "id" : "9tLCF4omb9hWmVXHwbj4CkiTkNDyTbC9xCmw65D1MKS3",
  "order2" : {
    "version" : 2,
    "id" : "DB2eX7VHJSkBvC95FYHWgEvHj47EsReHN4LQnD1EMLnb",
    "sender" : "3PQxNpso2uNbiPM7PQWJMNeYkVsUv4P5mLm",
    "senderPublicKey" : "Cb92PCaMeimQmmK1dSaoUro3j8YhskKhH89JWdmsCkQX",
    "matcherPublicKey" : "E3UwaHCQCySghK3zwNB8EDHoc3b8uhzGPFz3gHmWon4W",
    "assetPair" : {
      "amountAsset" : "4LHHvYGNKJUg5hj65aGD5vgScvCBmLpdRFtjokvCjSL8",
      "priceAsset" : null
    },
    "orderType" : "sell",
    "amount" : 113898170,
    "price" : 11636184,
    "timestamp" : 1572775953909,
    "expiration" : 1575281553909,
    "matcherFee" : 700000,
    "signature" : "2SY4WTcKVqLX97jnLC2iGghv2Q759S1HYWkHqueNUcVnSAH2YHt7F4Lr5onAHRA5LkJMAHYjw8byGiUmNG14ixXc",
    "proofs" : [ "2SY4WTcKVqLX97jnLC2iGghv2Q759S1HYWkHqueNUcVnSAH2YHt7F4Lr5onAHRA5LkJMAHYjw8byGiUmNG14ixXc" ]
  },
  "order1" : {
    "version" : 2,
    "id" : "HfK4b193jVSLuSZNFbgz1cSdN2qa1HkMJvEB86dFQCDY",
    "sender" : "3PAn2voDZPPiqgXPXNYENYit8PeEgGqey2A",
    "senderPublicKey" : "4EdEM9rL9mjZn3drNnkceH6DFVdYUgPx9jEmHN9aqEiw",
    "matcherPublicKey" : "E3UwaHCQCySghK3zwNB8EDHoc3b8uhzGPFz3gHmWon4W",
    "assetPair" : {
      "amountAsset" : "4LHHvYGNKJUg5hj65aGD5vgScvCBmLpdRFtjokvCjSL8",
      "priceAsset" : null
    },
    "orderType" : "buy",
    "amount" : 8544502410,
    "price" : 11682774,
    "timestamp" : 1573040813651,
    "expiration" : 1575546413651,
    "matcherFee" : 300000,
    "signature" : "3s2NmG4DzvwtXB3ZFm9JsYUUzH5ZV3qDhHeCdCrXCcZxf4z74PADf8AMogWdkU2GKda7WKJrNZ68jKiis2gmXzJR",
    "proofs" : [ "3s2NmG4DzvwtXB3ZFm9JsYUUzH5ZV3qDhHeCdCrXCcZxf4z74PADf8AMogWdkU2GKda7WKJrNZ68jKiis2gmXzJR" ]
  },
  "buyMatcherFee" : 3998,
  "timestamp" : 1573040813705
})
2019-11-06 12:47:14,930 INFO  [nio-worker-group-3-14] c.w.n.HandshakeHandler$Client - [53c00d33 34.253.153.4:6868] Accepted handshake Handshake(wavesW,(1,1,5),waves-official-eu-node-2,553497,Some(/34.253.153.4:6868))
2019-11-06 12:47:16,017 ERROR [nio-worker-group-3-14] c.w.n.RxExtensionLoader$ - Error requesting signatures: [id: 0x53c00d33, L:0.0.0.0/0.0.0.0:47436 ! R:/34.253.153.4:6868]
java.nio.channels.ClosedChannelException: null
	at io.netty.channel.AbstractChannel$AbstractUnsafe.write(...)(Unknown Source)
2019-11-06 12:47:23,315 INFO  [nio-worker-group-3-11] c.w.n.HandshakeHandler$Client - [508a74cc 52.50.69.247:6868] Accepted handshake Handshake(wavesW,(1,1,5),waves-official-eu-node-5,261901,Some(/52.50.69.247:6868))
2019-11-06 12:47:24,395 ERROR [nio-worker-group-3-11] c.w.n.RxExtensionLoader$ - Error requesting signatures: [id: 0x508a74cc, L:0.0.0.0/0.0.0.0:34076 ! R:/52.50.69.247:6868]
java.nio.channels.ClosedChannelException: null
	at io.netty.channel.AbstractChannel$AbstractUnsafe.write(...)(Unknown Source)


```

**Comment by vbonini** [comment]
[waves_stage.txt](https://github.com/wavesplatform/Waves/files/3974319/waves_stage.txt)

Additional logs as of this morning. Waves on stage is not progressing, and cannot serve as a failover node. Prod is unaffected at the moment.

**Comment by ksemaev** [comment]
I keep getting this constantly on all of our waves nodes. The only solution is resync from 0 which takes days. Please provide instructions on how to fix that situation

**Comment by ksemaev** [comment]
Same on v1.1.6  How do you handle forks?

**Comment by ksemaev** [comment]
Same on 1.1.7

**Comment by phearnot** [comment]
We are currently investigating this issue. Unfortunately, we don't have a working fix at this time.

There's a workaround for this issue:
1. Import blockchain from file.
1. Before running your node, backup the contents of `data` directory.
1. Run your node as usual. If your node forks or gets stuck, do the following:
    1. Replace your node's data directory with the one you backed up in step 2.
    1. Import blockchain from the most recent file. This process will complete much faster than importing blockchain into the empty state, as the node will process only the most recent blocks (the ones which were mined after you completed step 2)
    1. Proceed to step 2 (update your backup)

**Comment by Caveman2019** [comment]
Got the same error when trying to import from file

![image](https://user-images.githubusercontent.com/9812320/75387950-f4968580-5898-11ea-8a19-ae23fbf8015b.png)

**Comment by Caveman2019** [comment]
I am deleting the data folder and start over. :(

**Comment by phearnot** [comment]
This has been fixed in [1.1.11](https://github.com/wavesplatform/Waves/releases/tag/v1.1.11)
